src/service/data-util.service.ts

```diff
@@ -124,15 +125,21 @@ export class JhiDataUtils {
         return size.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ' ') + ' bytes';
     }
 
-    setFileData(event, entity, field: string, isImage: boolean) {
+    setFileData(event, entity, field: string, isImage: boolean, cb?: Function) {
         if (event && event.target.files && event.target.files[0]) {
             const file = event.target.files[0];
             if (isImage && !/^image\//.test(file.type)) {
+                if (cb) {
+                    cb();
+                }
                 return;
             }
             this.toBase64(file, (base64Data) => {
                 entity[field] = base64Data;
                 entity[`${field}ContentType`] = file.type;
+                if (cb) {
+                    cb();
```

Comment:
shouldn't this return updated entity to caller?

Reply:
previous implementation doesn't return anything.
My change was just to allow accepting and executing a callback passed by the caller